client/src/services/cryptpad.ts

```diff
@@ -164,7 +180,49 @@ export class Cryptpad {
       throw new CryptpadOpenError(CryptpadErrorCode.DocumentTypeNotEnabled, config.documentType);
     }
 
-    (window as any).CryptPadAPI(this.containerElement.id, { ...config });
+    // Set up a loading timeout (30 seconds) to detect stuck/corrupted files
+    const LOADING_TIMEOUT_MS = 30000;
+    this.clearLoadingTimeout();
+
+    // Create a promise that will be resolved when CryptPad successfully loads
+    let resolveLoading: (() => void) | null = null;
+    const loadingComplete = new Promise<void>((resolve) => {
+      resolveLoading = resolve;
+    });
+
+    const timeoutPromise = new Promise<void>((_, reject) => {
+      this.loadingTimeoutId = window.setTimeout(() => {
+        window.electronAPI.log('error', 'CryptPad loading timeout - file may be corrupted or too large');
+        reject(new Error('CryptPad loading timeout - the file may be corrupted, too large, or the server is not responding'));
```

Comment:
IMO a loading timeout shouldn't be an error.

Also, log and error messages are not consistent, shouldn't it be the same?
```suggestion
        const message = 'CryptPad loading timeout. The file may be corrupted or too large';
        window.electronAPI.log('error', message);
        reject(new Error(message));
```
```suggestion
        const message = 'CryptPad loading timeout. The file may be corrupted, too large, or the server is not responding';
        window.electronAPI.log('error', message);
        reject(new Error(message));
```